mark NemoEmbeddings as deprecated

raspawar opened the pull request:
The NemoEmbeddings is deprecated, instead use langchain-nvidia-ai-endpoints NVIDIAEmbeddings interface.

cc: @mattf

raspawar committed:
mark NemoEmbeddings as deprecated

dglogo committed:
docs: Update Nvidia documentation (#21240)

Updating Nvidia docs ahead for 5/15 competition. 

Thanks!

andyjessen committed:
docs: Fix comment within "handle long text" example (#21248)

The current doc-string comment is referring to the wrong schema.

baskaryan commented on libs/community/langchain_community/embeddings/nemo.py:
```diff
@@ -40,14 +41,22 @@ def is_endpoint_live(url: str, headers: Optional[dict], payload: Any) -> bool:
         raise Exception(f"Error querying the endpoint: {e}")
 
-
 class NeMoEmbeddings(BaseModel, Embeddings):
```
could we use the @deprecated decorator instead? see [example](https://github.com/langchain-ai/langchain/blob/64e17bd793483b63092407cf79d434179436b250/libs/community/langchain_community/chat_models/openai.py#L147)

SecurityViber committed:
Hotfix RetrievalQA Docs: docs: Fix formatting (#21183)

# Newline Characters breaking formatting 

**Description**: 
As you can see in the image below, the formatting in the documentation
is broken. As far as I can see the two added `\n` characters are
breaking the documentation. Therefore I would propose to remove those

![image](https://github.com/langchain-ai/langchain/assets/88305668/23b6e726-71b2-4812-91ea-3e8600683733)

**Dependencies**:
None

**Twitter Handle**
- epu9byj

---------

Co-authored-by: gere <gere@kapo.zh.ch>
Co-authored-by: Chester Curme <chester.curme@gmail.com>

mattf commented on libs/community/langchain_community/embeddings/nemo.py:
```diff
@@ -40,14 +41,22 @@ def is_endpoint_live(url: str, headers: Optional[dict], payload: Any) -> bool:
 
+    @root_validator(pre=True)
+    def raise_deprecation(cls, values: Dict) -> Dict:
+        print(values)
```
don't include print() statements

mattf commented on libs/community/langchain_community/embeddings/nemo.py:
```diff
@@ -6,6 +6,7 @@
 
 import aiohttp
 import requests
+import warnings
```
make sure you `poetry install --with dev,lint,typing,test` and run `make format lint test`

mattf reviewed (commented):
thank you for the contribution

i've verified the deprecation notice is visible from
- an app (`python use-nemo-embeddings.py`)
- python repl
- ipython repl

raspawar committed:
update the import and remove print staatement

raspawar committed:
Merge branch 'langchain-ai:master' into depr_nemo

raspawar committed:
update deprecation method

ccurme committed:
update
